modules/free/src/main/scala/doobie/free/kleisliinterpreter.scala

```diff
@@ -73,12 +74,12 @@ import doobie.free.callablestatement.{ CallableStatementIO, CallableStatementOp
 import doobie.free.resultset.{ ResultSetIO, ResultSetOp }
 
 object KleisliInterpreter {
-  def apply[M[_]: WeakAsync](logHandler: LogHandler[M]): KleisliInterpreter[M] =
+  def apply[M[_]: Async](logHandler: LogHandler[M]): KleisliInterpreter[M] =
```

Comment:
I feel that upgrading `WeakAsync` to more powerful `Async` here can break a lot of client code. So it may cause quite a turmoil. There are a lot of dependencies and assumptions across the library towards `WeakAsync` typeclass already. I would rather try to reconsider that fiber start/join approach below, especially because it does not seem absolutely necessary to me.

Reply:
I don't think it's possible to solve this problem in another way. However, I have some ideas on how to minimize compatibility issues. I'll try to implement a fix soon.

Reply:
Done with optional parameter primitive